### ISSUE TITLE
Batman: Vengeance, Batman: Rise of Sin Tzu (Xbox) - unsupported SM2/SS2/LM2/LS2 files

### ISSUE THREAD

**Issue by AnonBaiter** [losnoco/vgmstream#401]
sorry, i didn't mean to post all of this crap but since this is the DARE engine and the MAPS file takes care of STRM/STREAM files i had no choice
[[batman_vengeance]sp2_sm2[not_including_crucial_strm_file_because_my_upload_speed_is_horrendous].7z](https://drive.google.com/file/d/1L1hAooicICWItflh1olkDskNqAEtZO3W/view?usp=sharing)
[[batman_rise_of_sin_tzu]sp2_sm2.7z](https://drive.google.com/file/d/1TBxzSvXRijbsiTmMqAnM4a-iJXmpzU5q/view?usp=sharing)
[[batman_rise_of_sin_tzu]maps_lm2[english_language].7z](https://drive.google.com/file/d/1WUfyLE4asrP4DjNLgQrLuqbHhUMYLcGC/view?usp=sharing)

do note however that i haven't included said STRM/STREAM files (yet) because my upload speed is super slow sometimes, the second post of this issue focuses only on these files

**Comment by AnonBaiter** [comment]
the second post is dedicated to all these separate .7z files i'd said will be uploaded whenever possible
[[batman_vengeance]strm_ss2.7z](https://drive.google.com/file/d/1lTGdcX663-X620TFELPs3QM5EDyA_2dg/view?usp=sharing)
**_RESERVED_**
**_RESERVED_**

**Comment by bnnm** [comment]
Added Rise of Sin Tsu, though some subsongs fail, they are mono+stereo sequences I can't fix ATM (their individual subsongs are playable though).

Most Vengeance subsongs use the "Ubi ADPCM" codec that must be researched and added first.

**Comment by AnonBaiter** [comment]
> Most Vengeance subsongs use the "Ubi ADPCM" codec that must be researched and added first.

well, i _could_ research it somehow

**Comment by AnonBaiter** [comment]
> 
> 
> Added Rise of Sin Tsu, though some subsongs fail, they are mono+stereo sequences I can't fix ATM (their individual subsongs are playable though).
> 
> Most Vengeance subsongs use the "Ubi ADPCM" codec that must be researched and added first.

**Comment by AnonBaiter** [comment]
sorry for that i just closed the issue by accident

**Comment by bnnm** [comment]
This Ubi ADPCM comes in two modes, 6-bit (for voices) and 4-bit (for music). I got the 6-bit part, but the 4-bit is a lot more complex, since the code is using MMX that is pretty hard to follow.

**Comment by bnnm** [comment]
Works now.

**Comment by AnonBaiter** [comment]
thanks for making this work. i really appreciate it for reals.
